src/ReturnTypes/EloquentBuilderExtension.php

```diff
@@ -35,7 +42,7 @@ public function getClass(): string
     public function isMethodSupported(MethodReflection $methodReflection): bool
     {
         // Don't handle dynamic wheres
-        if (Str::startsWith($methodReflection->getName(), 'where')) {
+        if (Str::startsWith($methodReflection->getName(), 'where') && ! in_array($methodReflection->getName(), self::NON_DYNAMIC_WHERES, true)) {
             return false;
         }
```

Comment:
```suggestion
        $builderReflection = $this->getBroker()->getClass(EloquentBuilder::class);

        // Don't handle dynamic wheres
        if (Str::startsWith($methodReflection->getName(), 'where') &&
            ! $builderReflection->hasNativeMethod($methodReflection->getName())
        ) {
            return false;
        }
```

Instead of checking if the method name is in the array, we can check if builder has that method. This is more future-proof.